packages/now-routing-utils/src/superstatic.ts

```diff
@@ -181,7 +183,24 @@ function replaceSegments(
       indexes[name] = toSegmentDest(index);
     });
 
+    let destParams: string[] = [];
+
     if (destination.includes(':') && segments.length > 0) {
+      const pathnameKeys: Key[] = [];
+      const hashKeys: Key[] = [];
+
+      try {
+        pathToRegexp(pathname, pathnameKeys);
+        pathToRegexp(hash || '', hashKeys);
```

Comment:
Do we also need to consider other parts of the URL such as the query string?

Reply:
I guess thats already accounted for since we wont add duplicate query string keys

Reply:
It might be good to also opt-out of auto-appending if they're used in the destination's query 🤔 we could investigate adding this behavior in follow-up PRs